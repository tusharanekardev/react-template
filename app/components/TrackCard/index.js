/**
 *
 * TrackCard
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

// import PropTypes from 'prop-types'
// import styled from 'styled-components'
import styled from '@emotion/styled';
import { Card } from '@mui/material';

const CustomCard = styled(Card)`
  && {
    margin: 1rem 0;
    padding: 1rem;
  }
`;

/**
 * MediaItemCard component that displays information about a Media item (like a song).
 * It shows the repository's name, full name, and star count.
 *
 * @date 01/03/2024 - 14:47:28
 *
 * @param {Object} props - The component props.
 * @param {string} props.trackName - The name of the track.
 * @param {string} props.collectionName - The name of the collection.
 * @param {string} props.artistName - The name of the artist.
 * @param {string} props.country - The name of the country the media is from
 * @param {string} props.primaryGenreName - Genre name
 * @param {string} props.thumbnailSrc - thumbnail of the media
 * @returns {JSX.Element} The RepoCard component displaying the repository information.
 */
export function TrackCard({ artistName, collectionName, previewUrl, artworkUrl60 }) {
  return (
    <CustomCard data-testid="track-card">
      <p>{artistName}</p>
      <p>{collectionName}</p>
      <p>{artistName}</p>
      <p>{artistName}</p>
    </CustomCard>
  );
}

TrackCard.propTypes = {
  artistName: PropTypes.string,
  collectionName: PropTypes.string,
  previewUrl: PropTypes.string,
  artworkUrl60: PropTypes.string
};

export default TrackCard;
