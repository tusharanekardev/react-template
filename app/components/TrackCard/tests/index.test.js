/**
 *
 * Tests for TrackCard
 *
 */

import React from 'react';
import { renderProvider } from '@utils/testUtils';
import TrackCard from '../index';

describe('<TrackCard />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<TrackCard />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 TrackCard component', () => {
    const { getAllByTestId } = renderProvider(<TrackCard />);
    expect(getAllByTestId('track-card').length).toBe(1);
  });
});
