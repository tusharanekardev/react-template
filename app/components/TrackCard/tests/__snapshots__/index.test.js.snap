// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root e12nrugs0 css-16zkfrb-MuiPaper-root-MuiCard-root"
      data-testid="track-card"
    >
      <p />
      <p />
      <p />
      <p />
    </div>
  </div>
</body>
`;
